team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/TextMergeViewer.java

```diff
@@ -5408,6 +5410,32 @@ private boolean isIgnoreAncestor() {
 
 		updateVScrollBar();
 		updatePresentation();
+		updateToolbarLabel();
+	}
+
+	private void updateToolbarLabel() {
+		final String DIFF_COUNT_ID = "DiffCount"; //$NON-NLS-1$
+		ToolBarManager tbm =
+				(ToolBarManager) getToolBarManager(fComposite.getParent());
+		int differenceCount = fMerger.changesCount();
+		if (tbm != null) {
+
+			String label = differenceCount > 1 ? differenceCount + " Differences" //$NON-NLS-1$
+					: differenceCount == 1 ? differenceCount + " Difference" : "No Difference"; //$NON-NLS-1$ //$NON-NLS-2$
+			LabelContributionItem labelContributionItem = new LabelContributionItem(DIFF_COUNT_ID,
+					label);
+
+			if (tbm.find(DIFF_COUNT_ID) != null) {
+				tbm.replaceItem(DIFF_COUNT_ID, labelContributionItem);
+			} else {
+				tbm.appendToGroup("diffLabel", labelContributionItem); //$NON-NLS-1$
+			}
+			Display.getDefault().asyncExec(() -> {
```

Comment:
Some issues: 
1) display should be better retrieved from the widget on which we are operating
2) in all `asyncExec()` cals first thing to do is to check if the UI we are referring to is still there.
3) `setRedraw(true)` is wrong. I guess you wanted to use `control.getParent().redraw();` but also this is questionable & I don't see a need for this.

So please replace this with
```
			fComposite.getDisplay().asyncExec(() -> {
				// relayout in next tick
				ToolBar control = tbm.getControl();
				if (control != null && !control.isDisposed()) {
					tbm.update(true);
				}
			});
```

Reply:
To be in sync the below code was taken from InternalRefresh Method of ContentMergeViewer.java 

`Display.getDefault().asyncExec(() -> {`
						`// relayout in next tick`
						`tbm.update(true);`
						`tbm.getControl().getParent().setRedraw(true);`
					`});`
which was fixed for "bad UI on MacOS merge viewer" . Should it be replaced with below code here now as it is changed there ?
`tbm.update(true);`
`tbm.getControl().getParent().layout(true);`

Reply:
Please check if proposed code works on Windows. If yes, no need to copy/paste code you mention, especially we know it doesn't work properly on Mac (this is the reason it is changed back and forth).

Reply:
The changes are adopted . Please have a look